Kernel/Devices/VMWareSVGADevice.cpp

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2021, the Serenity OS Developers
+ * Copyright (c) 2021, Alexander Richards <electrodeyt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "VMWareSVGADevice.h"
+#include <AK/Singleton.h>
+#include <Kernel/PCI/Access.h>
+#include <Kernel/VM/AnonymousVMObject.h>
+#include <Kernel/VM/MemoryManager.h>
+#include <LibC/errno_numbers.h>
+#include <LibC/sys/ioctl_numbers.h>
+
+namespace Kernel {
+
+static AK::Singleton<VMWareSVGADevice> s_the;
+
+void VMWareSVGADevice::initialize()
+{
+    s_the.ensure_instance();
+}
+
+VMWareSVGADevice& VMWareSVGADevice::the()
+{
+    return *s_the;
+}
+
+// We put a fake IRQ into the IRQ Handler, as we will change it anyway
+// TODO: this is a really nasty hack that only works as the IRQ is registered on the first call to enable_irq()
+// This should probably be passed in by the init_stage2
+VMWareSVGADevice::VMWareSVGADevice()
+    : BlockDevice(29, 0)
+    , IRQHandler(0)
+{
+    dbgln("VMWareSVGA: init");
+    // Find the VMWare graphics
+    static const PCI::ID vmware_id = { 0x15AD, 0x0405 };
```

Comment:
might be worth it to add this vendor id to PCI/IDs.h (and use it in init2 as well :)